sources/packages/web/src/components/common/Root.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div
+    class="v-main body-background"
+    @mouseover="setLastActivityTime"
+    @click="setLastActivityTime"
+    @keyup="setLastActivityTime"
+  >
+    <slot></slot>
+    <ConfirmExtendTime
+      ref="extendTimeModal"
+      :startTimer="startTimer"
+      :clientIdType="clientType"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { useRoute } from "vue-router";
+import { onMounted, ref, onUnmounted, computed } from "vue";
+import { ClientIdType, AppRoutes } from "@/types";
+import { useAuth, ModalDialog, useClientLoader } from "@/composables";
+import {
+  MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER,
+  MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT,
+  MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION,
+  MINIMUM_IDLE_TIME_FOR_WARNING_AEST,
+} from "@/constants/system-constants";
+import ConfirmExtendTime from "@/components/common/modals/ConfirmExtendTime.vue";
+
+export default {
+  components: { ConfirmExtendTime },
+  props: {
+    clientIdType: {
+      type: String,
+      required: true,
+      default: "" as ClientIdType,
+    },
+  },
+  setup(props: any) {
+    const { executeRenewTokenIfExpired } = useAuth();
+    const route = useRoute();
+    const lastActivityLogin = ref(new Date());
+    const interval = ref();
+    const extendTimeModal = ref({} as ModalDialog<boolean>);
+    const startTimer = ref(false);
+    const { getClientType } = useClientLoader();
+
+    const clientType = computed(() => {
+      return getClientType(props.clientIdType);
+    });
+
+    const minimumIdleTime = computed(() => {
```

Comment:
Here maybe we can use a similar approach to the one that we are using to get the Keycloak client name, as shown below. In this way, we do not need the 4 constants and there is no need to check each one in a switch.

![image](https://user-images.githubusercontent.com/61259237/139491603-027ec23a-47a6-4585-9ca5-01697cfe0d6a.png)

Reply:
added constant because of this task,
![image](https://user-images.githubusercontent.com/77353155/139492514-6aec1580-ebb1-4954-adad-2085ad9445f7.png) and I confirmed with @abschwenker then added separate constants for each portal 
![image](https://user-images.githubusercontent.com/77353155/139492693-257ae9e6-30af-424f-84c5-5b65cf5f9089.png)

Reply:
The way that I mentioned will allow the same configuration per client in the same way that the constants provided.